zuul-core/src/main/java/com/netflix/zuul/netty/connectionpool/DefaultClientChannelManager.java

```diff
@@ -100,55 +103,49 @@ public class DefaultClientChannelManager implements ClientChannelManager {
 
     public static final String IDLE_STATE_HANDLER_NAME = "idleStateHandler";
 
-    public DefaultClientChannelManager(OriginName originName, IClientConfig clientConfig, Registry spectatorRegistry) {
-        this(originName, clientConfig, new DynamicServerResolver(clientConfig), spectatorRegistry);
+    public DefaultClientChannelManager(OriginName originName, IClientConfig clientConfig, Registry registry) {
+        this(originName, clientConfig, new DynamicServerResolver(clientConfig), registry);
     }
 
     public DefaultClientChannelManager(
-            OriginName originName,
-            IClientConfig clientConfig,
-            Resolver<DiscoveryResult> resolver,
-            Registry spectatorRegistry) {
+            OriginName originName, IClientConfig clientConfig, Resolver<DiscoveryResult> resolver, Registry registry) {
         this.originName = Objects.requireNonNull(originName, "originName");
         this.dynamicServerResolver = resolver;
 
-        String metricId = originName.getMetricId();
-
         this.clientConfig = clientConfig;
-        this.spectatorRegistry = spectatorRegistry;
+        this.registry = registry;
         this.perServerPools = new ConcurrentHashMap<>(200);
 
         this.connPoolConfig = new ConnectionPoolConfigImpl(originName, this.clientConfig);
 
-        this.createNewConnCounter = SpectatorUtils.newCounter(METRIC_PREFIX + "_create", metricId);
-        this.createConnSucceededCounter = SpectatorUtils.newCounter(METRIC_PREFIX + "_create_success", metricId);
-        this.createConnFailedCounter = SpectatorUtils.newCounter(METRIC_PREFIX + "_create_fail", metricId);
-
-        this.closeConnCounter = SpectatorUtils.newCounter(METRIC_PREFIX + "_close", metricId);
-        this.closeAbovePoolHighWaterMarkCounter =
-                SpectatorUtils.newCounter(METRIC_PREFIX + "_closeAbovePoolHighWaterMark", metricId);
-        this.closeExpiredConnLifetimeCounter =
-                SpectatorUtils.newCounter(METRIC_PREFIX + "_closeExpiredConnLifetime", metricId);
-        this.requestConnCounter = SpectatorUtils.newCounter(METRIC_PREFIX + "_request", metricId);
-        this.reuseConnCounter = SpectatorUtils.newCounter(METRIC_PREFIX + "_reuse", metricId);
-        this.releaseConnCounter = SpectatorUtils.newCounter(METRIC_PREFIX + "_release", metricId);
-        this.alreadyClosedCounter = SpectatorUtils.newCounter(METRIC_PREFIX + "_alreadyClosed", metricId);
-        this.connTakenFromPoolIsNotOpen = SpectatorUtils.newCounter(METRIC_PREFIX + "_fromPoolIsClosed", metricId);
-        this.maxConnsPerHostExceededCounter =
-                SpectatorUtils.newCounter(METRIC_PREFIX + "_maxConnsPerHostExceeded", metricId);
-        this.closeWrtBusyConnCounter = SpectatorUtils.newCounter(METRIC_PREFIX + "_closeWrtBusyConnCounter", metricId);
+        this.createNewConnCounter = newCounter("_create");
```

Comment:
nit: move the `_` to the `newCounter` and `newGauge` functions so these all read a bit better 😅